src/core/config-printf.c

```diff
@@ -0,0 +1,64 @@
+/***
+  This file is part of systemd.
+
+  Copyright 2016 Red Hat, Inc.
+
+  systemd is free software; you can redistribute it and/or modify it
+  under the terms of the GNU Lesser General Public License as published by
+  the Free Software Foundation; either version 2.1 of the License, or
+  (at your option) any later version.
+
+  systemd is distributed in the hope that it will be useful, but
+  WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public License
+  along with systemd; If not, see <http://www.gnu.org/licenses/>.
+***/
+
+#include <errno.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "formats-util.h"
+#include "config-printf.h"
+#include "macro.h"
+#include "specifier.h"
+#include "unit-name.h"
+#include "user-util.h"
+
+int config_printf(const char *format, char **ret) {
+
+        /* This is similar to unit_full_printf() but only supports
+         * specifiers that aren't specific to units
+         *
+         * %t the runtime directory to place sockets in (e.g. "/run" or $XDG_RUNTIME_DIR)
+         * %U the UID of the running user
+         * %u the username of the running user
+         * %h the homedir of the running user
+         * %s the shell of the running user
+         * %m the machine ID of the running system
+         * %H the host name of the running system
+         * %b the boot ID of the running system
+         * %v `uname -r` of the running system
+         */
+
+        const Specifier table[] = {
+                { 'U', specifier_user_id,        NULL },
+                { 'u', specifier_user_name,      NULL },
+                { 'h', specifier_user_home,      NULL },
+                { 's', specifier_user_shell,     NULL },
+                { 'm', specifier_machine_id,     NULL },
+                { 'H', specifier_host_name,      NULL },
+                { 'b', specifier_boot_id,        NULL },
+                { 'v', specifier_kernel_release, NULL },
+                {}
```

Comment:
This will also require an addition to the documentation... systemd-user.conf.xml could just say: "a subset of specifiers are expanded (%U, %u, %h, …). See systemd.unit(5) for their meanings."